hazelcast-jet-hadoop/src/test/java/com/hazelcast/jet/connector/hadoop/ReadHdfsPTest.java

```diff
@@ -46,67 +60,102 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-@Category(QuickTest.class)
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelTest.class})
 public class ReadHdfsPTest extends JetTestSupport {
 
-    @Test
-    public void testReadFile() throws Exception {
-        Path path = writeToFile("key-1 value-1\n", "key-2 value-2\n", "key-3 value-3\n", "key-4 value-4\n");
+    private static final String[] ENTRIES = {"key-1 value-1\n", "key-2 value-2\n", "key-3 value-3\n", "key-4 value-4\n"};
 
-        JetInstance instance = createJetMember();
-        createJetMember();
-        DAG dag = new DAG();
-        Vertex source = dag.newVertex("source", readHdfs(path.toString()))
-                           .localParallelism(4);
-        Vertex sink = dag.newVertex("sink", writeList("sink"))
-                         .localParallelism(1);
-        dag.edge(between(source, sink));
+    @Parameterized.Parameter(0)
+    public Class<? extends InputFormat> inputFormatClass;
 
-        Future<Void> future = instance.newJob(dag).execute();
-        assertCompletesEventually(future);
+    @Parameterized.Parameter(1)
+    public Distributed.BiFunction mapper;
 
+    @Parameterized.Parameters(name = "Executing: {0} {1}")
+    public static Collection<Object[]> parameters() {
+        Distributed.BiFunction defaultMapper = Util::entry;
+        Distributed.BiFunction mapper = (k, v) -> v.toString();
 
-        IList<Map.Entry> list = instance.getList("sink");
-        assertEquals(4, list.size());
-        assertTrue(list.get(0).getValue().toString().contains("value"));
+        return Arrays.asList(
+                new Object[]{TextInputFormat.class, defaultMapper}, //
+                new Object[]{TextInputFormat.class, mapper}, //
+                new Object[]{SequenceFileInputFormat.class, defaultMapper}, //
+                new Object[]{SequenceFileInputFormat.class, mapper} //
+        );
     }
 
-    @Test
-    public void testReadFile_withMapping() throws Exception {
-        Path path = writeToFile("key-1 value-1\n", "key-2 value-2\n", "key-3 value-3\n", "key-4 value-4\n");
+    private JobConf jobConf;
 
-        JetInstance instance = createJetMember();
+    private JetInstance instance;
+
+    @Before
+    public void setup() throws IOException {
+        instance = createJetMember();
         createJetMember();
+
+        jobConf = new JobConf();
+        jobConf.setInputFormat(inputFormatClass);
+
+        Path path = writeToFile();
+        FileInputFormat.addInputPath(jobConf, path);
```

Comment:
can we make the examples read more than one file at a time, this is a common case with HDFS.